component/common/loki/wal/watcher.go

```diff
@@ -150,10 +184,27 @@ func (w *Watcher) run() error {
 	return nil
 }
 
+// findSegmentForIndex finds the first segment greater than or equal to index.
+func (w *Watcher) findSegmentForIndex(index int) (int, error) {
+	// TODO(thepalbi): is segs in order?
+	segs, err := readSegmentNumbers(w.walDir)
+	if err != nil {
+		return -1, err
+	}
+
+	for _, r := range segs {
+		if r > index {
+			return r, nil
+		}
+	}
+
+	return -1, errors.New("failed to find segment for index")
+}
+
 // watch will start reading from the segment identified by segmentNum. If an EOF is reached, it will keep
 // reading for more WAL records with a wlog.LiveReader. Periodically, it will check if there's a new segment, and if positive
 // read the remaining from the current one and return.
-func (w *Watcher) watch(segmentNum int) error {
+func (w *Watcher) watch(segmentNum int, tail bool) error {
```

Comment:
It'd be good to mention in the comment above what `tail` is for.

Reply:
Clarified